bot/seasons/season.py

```diff
@@ -38,48 +43,79 @@ def get_season(bot, season_name: str = None, date: datetime.date = None):
 
     seasons = get_seasons()
 
-    # If there's a season name, we can just grab the correct class.
+    # Use season override if season name not provided
+    if not season_name and Client.season_override:
+        log.debug(f"Season override found: {Client.season_override}")
+        season_name = Client.season_override
+
+    # If name provided grab the specified class or fallback to evergreen.
     if season_name:
         season_name = season_name.lower()
         if season_name not in seasons:
             season_name = 'evergreen'
-
-        season_lib = importlib.import_module(f'bot.seasons.{season_name}')
-        season_class = getattr(season_lib, season_name.capitalize())
+        season_class = get_season_class(season_name)
         return season_class(bot)
 
     # If not, we have to figure out if the date matches any of the seasons.
     seasons.remove('evergreen')
-
     for season_name in seasons:
-        season_lib = importlib.import_module(f'bot.seasons.{season_name}')
-        season_class = getattr(season_lib, season_name.capitalize())
-        season_object = season_class(bot)
-        if season_object.start <= date <= season_object.end:
-            return season_object
+        season_class = get_season_class(season_name)
+        # check if date matches before returning an instance
+        if season_class.start() <= date <= season_class.end():
+            return season_class(bot)
     else:
-        evergreen_lib = importlib.import_module(f'bot.seasons.evergreen')
-        return evergreen_lib.Evergreen(bot)
+        evergreen_class = get_season_class('evergreen')
+        return evergreen_class(bot)
 
 
 class SeasonBase:
     name = None
+    date_format = "%d/%m-%Y"
+
+    @staticmethod
+    def current_year():
+        return datetime.date.today().year
 
-    def __init__(self):
-        current_year = datetime.date.today().year
-        date_format = "%d/%m-%Y"
-        self.start = datetime.datetime.strptime(f"{self.start_date}-{current_year}", date_format).date()
-        self.end = datetime.datetime.strptime(f"{self.end_date}-{current_year}", date_format).date()
+    @classmethod
+    def start(cls):
+        return datetime.datetime.strptime(f"{cls.start_date}-{cls.current_year()}", cls.date_format).date()
+
+    @classmethod
+    def end(cls):
+        return datetime.datetime.strptime(f"{cls.end_date}-{cls.current_year()}", cls.date_format).date()
+
+    @staticmethod
```

Comment:
While functionally the same, I do think a static method is more appropriate for these.